src/io_posix.c

```diff
@@ -40,5 +40,5 @@ udx__recvmsg (udx_socket_t *handle, uv_buf_t *buf, struct sockaddr *addr, int ad
     size = recvmsg(handle->io_poll.io_watcher.fd, &h, 0);
   } while (size == -1 && errno == EINTR);
 
-  return size == -1 ? uv_translate_sys_error(errno) : size;
+  return size == -1 ? (errno == EWOULDBLOCK ? 0 : uv_translate_sys_error(errno)) : size;
```

Comment:
```suggestion
  return size == -1 ? (errno == EAGAIN ? 0 : uv_translate_sys_error(errno)) : size;
```

> Macro: _int_ *EWOULDBLOCK*
> “Operation would block.” In the GNU C Library, this is another name for EAGAIN (above). The values are always the same, on every operating system.
>
> C libraries in many older Unix systems have EWOULDBLOCK as a separate error code.

https://www.gnu.org/savannah-checkouts/gnu/libc/manual/html_node/Error-Codes.html

Reply:
libuv also only defines `UV_EAGAIN` for this reason.

Reply:
AH! I was trying to figure that out. So libuv translates EWOULDBLOCK and friends to UV_EAGAIN ?

Reply:
makes my loop much simpler if we can just use UV_EAGAIN

Reply:
I think it's up to us to translate `errno` to `EAGAIN` in case of `EWOULDBLOCK`; `uv_translate_sys_error()` just negates the passed `errno` if it's not already negated.

Reply:
AFAIK, it's only relevant to _really_ old systems though as all other systems define `EWOULDBLOCK` to be the same as `EAGAIN` 🙈

Reply:
seems like it's ancient, so mb i just make it with EAGAIN to keep it simple, wdyt?

Reply:
KISS 👍